sos/plugins/kubernetes.py

```diff
@@ -15,40 +15,130 @@
+    files = ("/etc/origin/master/master-config.yaml",)
+
+    option_list = [
+        ("all", "use --all-namespaces for all kubectl commands",
+            'fast', True),
+        ("describe", "capture descriptions of all kube resources",
+            'fast', True),
+        ("podlogs", "capture logs for pods", 'slow', False),
+    ]
+
+    def check_is_master(self):
+        if any([path.exists("/var/run/kubernetes/apiserver.key"),
+                path.exists("/etc/origin/master/master-config.yaml")
+                ]):
+            return True
+        else:
+            return False
 
     def setup(self):
         self.add_copy_spec("/etc/kubernetes")
         self.add_copy_spec("/var/run/flannel")
 
-        # Kubernetes master info
-        self.add_cmd_output("kubectl version")
-        self.add_cmd_output("kubectl get -o json pods")
-        self.add_cmd_output("kubectl get -o json nodes")
-        self.add_cmd_output("kubectl get -o json services")
-        self.add_cmd_output("kubectl get -o json replicationController")
-        self.add_cmd_output("kubectl get -o json events")
-        self.add_cmd_output("journalctl -u kubelet")
-        self.add_cmd_output("journalctl -u kube-apiserver")
-        self.add_cmd_output("journalctl -u kube-controller-manager")
-        self.add_cmd_output("journalctl -u kube-scheduler")
-        self.add_cmd_output("journalctl -u kube-proxy")
-
-        if self.get_option('podslog'):
-            result = self.get_command_output("kubectl get pods")
-            if result['status'] == 0:
-                for line in result['output'].splitlines()[1:]:
-                    pod_name = line.split(" ")[0]
-                    self.add_cmd_output([
-                        "{0} log {1}".format("kubectl", pod_name)
-                    ])
+        svcs = [
+            'kubelet', 'kube-apiserver', 'kube-proxy', 'kube-scheduler',
+            'kube-controller-manager'
+        ]
+
+        for svc in svcs:
+            self.add_cmd_output("journalctl -u {}".format(svc))
+
+        # We can only grab kubectl output from the master
+        if self.check_is_master():
+            kube_bin = "kubectl "
+            if path.exists('/etc/origin/master/admin.kubeconfig'):
+                kube_bin += "--config=/etc/origin/master/admin.kubeconfig"
+
+            kube_get = "get -o json "
+            if self.get_option('all'):
+                kube_get += "--all-namespaces=true"
+
+            for cmd in ['version', 'config view']:
+                self.add_cmd_output("{} {}".format(kube_bin, cmd))
+
+            resources = [
+                'events', 'limitrange', 'pods', 'pvc', 'rc',
+                'resourcequota', 'services'
+            ]
+
+            for res in resources:
+                self.add_cmd_output(
+                    "{} {} {}".format(
+                        kube_bin,
+                        kube_get,
+                        res
+                    )
+                )
```

Comment:
I think this would look better using a list comprehension:

```python
    self.add_cmd_output([
        "%s %s %s" % (kube_bin, kube_get, res) for res in resources
    ])
```

It looks a lot neater & produces the same result:

```python
>>> resources = ["foo", "bar", "baz", "qux" ]
>>> kube_bin = "kubctl --foo"
>>> kube_get = "get"
>>> list = [ "%s %s %s" % (kube_bin, kube_get, res) for res in resources ]
>>> list
['kubctl --foo get foo', 'kubctl --foo get bar', 'kubctl --foo get baz', 'kubctl --foo get qux']
```

(use `.format` if you must - we haven't set a hard standard for plugin authors to follow)